be2-scala/src/main/scala/ch/epfl/pop/pubsub/PublishSubscribe.scala

```diff
@@ -65,8 +84,12 @@ object PublishSubscribe {
         methodPartitioner.out(portPipelineError) ~> merger
         methodPartitioner.out(portParamsWithMessage) ~> hasMessagePartition ~> merger
         methodPartitioner.out(portParamsWithChannel) ~> hasChannelPartition ~> merger
+        methodPartitioner.out(portParamsWithMap) ~> hasMapPartition ~> merger
+        methodPartitioner.out(portResponseHandler) ~> responsePartition ~> merger
 
-        merger ~> jsonRpcAnswerGenerator ~> jsonRpcAnswerer ~> output
+        merger ~> broadcast
+        broadcast.out(0) ~> jsonRpcAnswerGenerator ~> jsonRpcAnswerer ~> output
+        broadcast.out(1) ~> monitorSink
```

Comment:
You could add variables such as `portBroadcastSuccess`, `portBroadcastFailure` and `totalBroadcastPorts` to remove magic numbers and make their intent clearer.

Reply:
These number have no meaning really, both pipe output the same data. I think Akka even lets me remove them, I will check that.

Reply:
taken care of in commit: a8f08f609d32e42aabbc2159fbdc8efab80f9602